lib/ansible/modules/cloud/amazon/batch_job_definition.py

```diff
@@ -0,0 +1,526 @@
+        maps to ReadonlyRootfs in the Create a container section of the Docker Remote API and the --read-only option
+        to docker run.
+
+  privileged:
+    description:
+      - When this parameter is true, the container is given elevated privileges on the host container instance
+        (similar to the root user). This parameter maps to Privileged in the Create a container section of the
+        Docker Remote API and the --privileged option to docker run.
+
+  ulimits:
+    description:
+      - A list of ulimits to set in the container. This parameter maps to Ulimits in the Create a container section
+        of the Docker Remote API and the --ulimit option to docker run. List of dictionaries.
+    suboptions:
+      hardLimit:
+        description:
+          - The hard limit for the ulimit type.
+      name:
+        description:
+          - The type of the ulimit.
+      softLimit:
+        description:
+          - The soft limit for the ulimit type.
+
+  user:
+    description:
+      - The user name to use inside the container. This parameter maps to User in the Create a container section of
+        the Docker Remote API and the --user option to docker run.
+
+  attempts:
+    description:
+      - Retry strategy - The number of times to move a job to the RUNNABLE status. You may specify between 1 and 10
+        attempts. If attempts is greater than one, the job is retried if it fails until it has moved to RUNNABLE that
+        many times.
+
+requirements:
+    - boto3
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+---
+- hosts: localhost
+  gather_facts: no
+  vars:
+    state: present
+  tasks:
+- name: My Batch Job Definition
+  batch_job_definition:
+    job_definition_name: My Batch Job Definition
+    state: present
+    type: container
+    parameters:
+      Param1: Val1
+      Param2: Val2
+    image: <Docker Image URL>
+    vcpus: 1
+    memory: 512
+    command:
+      - python
+      - run_my_script.py
+      - arg1
+    job_role_arn: <Job Role ARN>
+    attempts: 3
+  register: job_definition_create_result
+
+- name: show results
+  debug: var=job_definition_create_result
+'''
+
+RETURN = '''
+---
+batch_job_definition_action:
+    description: describes what action was taken
+    returned: success
+    type: string
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn, HAS_BOTO3
+from botocore.exceptions import ClientError, ParamValidationError, MissingParametersError
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   Helper Functions & classes
+#
+# ---------------------------------------------------------------------------------------------------
+
+# logger = logging.getLogger()
+# logging.basicConfig(filename='ansible_debug.log')
+# logger.setLevel(logging.DEBUG)
+
+
+class AWSConnection:
```

Comment:
Since this is only ever used once, I think it would be easier to read to just create the different client/resource connections and pass those in.